pkg/virtctl/vm/stop.go

```diff
@@ -0,0 +1,90 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package vm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+const COMMAND_STOP = "stop"
+
+func NewStopCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "stop (VM)",
+		Short:   "Stop a virtual machine.",
+		Example: usage(COMMAND_STOP),
+		Args:    templates.ExactArgs("stop", 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			vmOpt, err = BeforeRun(args, clientConfig)
+			if err != nil {
+				return err
+			}
+			c := Command{command: COMMAND_STOP, clientConfig: clientConfig}
+			return c.stopRun(args, cmd)
+		},
+	}
+
+	cmd.Flags().BoolVar(&forceRestart, forceArg, false, "--force=false: Only used when grace-period=0. If true, immediately remove VMI pod from API and bypass graceful deletion. Note that immediate deletion of some resources may result in inconsistency or data loss and requires confirmation.")
+	cmd.Flags().Int64Var(&gracePeriod, gracePeriodArg, -1, "--grace-period=-1: Period of time in seconds given to the VMI to terminate gracefully. Can only be set to 0 when --force is true (force deletion). Currently only setting 0 is supported.")
+	cmd.Flags().BoolVar(&dryRun, dryRunArg, false, dryRunCommandUsage)
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	return cmd
+}
+
+func (o *Command) stopRun(args []string, cmd *cobra.Command) error {
+	vmiName = args[0]
+
+	dryRunOption := getDryRunOption(dryRun)
+	if len(dryRunOption) > 0 && dryRunOption[0] == metav1.DryRunAll {
+		fmt.Printf("Dry Run execution\n")
+	}
+
+	if cmd.Flags().Changed(gracePeriodArg) && forceRestart == false {
+		return fmt.Errorf("Can not set gracePeriod without --force=true")
+	}
+
+	if forceRestart {
```

Comment:
The logic here is unnecessarily complex, can you fix that in a follow up?

Reply:
+1 definetly, here we need a refactor